lib/milc_interface.cpp

```diff
@@ -645,6 +645,169 @@ void qudaGaugeForce(int precision, int num_loop_types, double milc_loop_coeff[3]
   qudaGaugeForcePhased(precision, num_loop_types, milc_loop_coeff, eb3, arg, 0);
 }
 
+/**
+ * @brief Reusable routine that creates a qudaGaugeParam for gauge-related observable measurements
+ *
+ * @param[in] precision MILC precision
+ * @param[in] arg MILC Site arg structure
+ * @param[in] phase_in Whether or not phases have been applied
+ * @return A qudaGaugeParam that can be passed to QUDA interface functions
+ */
+QudaGaugeParam createGaugeParamForObservables(int precision, QudaMILCSiteArg_t *arg, int phase_in)
+{
+  QudaGaugeParam qudaGaugeParam = newMILCGaugeParam(localDim,
+      (precision==1) ? QUDA_SINGLE_PRECISION : QUDA_DOUBLE_PRECISION, QUDA_WILSON_LINKS);
+
+  qudaGaugeParam.gauge_offset = arg->link_offset;
+  qudaGaugeParam.mom_offset = arg->mom_offset;
+  qudaGaugeParam.site_size = arg->size;
+  qudaGaugeParam.gauge_order = arg->site ? QUDA_MILC_SITE_GAUGE_ORDER : QUDA_MILC_GAUGE_ORDER;
+  qudaGaugeParam.staggered_phase_applied = phase_in;
+  qudaGaugeParam.staggered_phase_type = QUDA_STAGGERED_PHASE_MILC;
+  // FIXME: phases and boundary conditions are "munged" together inside QUDA, so the unphase function
+  // doesn't change the boundary condition flag. This setting guarantees that phases and boundary conditions
+  // are consistently set under the hood --- but we still need an extra minus sign on the output.
+  qudaGaugeParam.t_boundary = QUDA_PERIODIC_T;
+  //if (phase_in) qudaGaugeParam.t_boundary = QUDA_ANTI_PERIODIC_T;
+  if (phase_in) qudaGaugeParam.reconstruct_sloppy = qudaGaugeParam.reconstruct = QUDA_RECONSTRUCT_NO;
+
+  qudaGaugeParam.ga_pad = 0;
+  // For multi-GPU, ga_pad must be large enough to store a time-slice
+#ifdef MULTI_GPU
+  int x_face_size = qudaGaugeParam.X[1] * qudaGaugeParam.X[2] * qudaGaugeParam.X[3] / 2;
+  int y_face_size = qudaGaugeParam.X[0] * qudaGaugeParam.X[2] * qudaGaugeParam.X[3] / 2;
+  int z_face_size = qudaGaugeParam.X[0] * qudaGaugeParam.X[1] * qudaGaugeParam.X[3] / 2;
+  int t_face_size = qudaGaugeParam.X[0] * qudaGaugeParam.X[1] * qudaGaugeParam.X[2] / 2;
+  int pad_size = x_face_size > y_face_size ? x_face_size : y_face_size;
+  pad_size = pad_size > z_face_size ? pad_size : z_face_size;
+  pad_size = pad_size > t_face_size ? pad_size : t_face_size;
+  qudaGaugeParam.ga_pad = pad_size;
+#endif
+
+  if (!have_resident_gauge) {
+    qudaGaugeParam.make_resident_gauge = false;
+    qudaGaugeParam.use_resident_gauge = false;
+  } else {
+    qudaGaugeParam.use_resident_gauge = true;
+    qudaGaugeParam.make_resident_gauge = true;
+  }
+
+  return qudaGaugeParam;
+}
+
+void qudaGaugeLoopTracePhased(int precision, double *traces, int** input_path_buf, int* path_length, double *loop_coeff, int num_paths,
```

Comment:
Wondering if we might want `qudaGaugeLoopTrace`, `qudaPlaquette`, `qudaPolyakovLoop` functions with the phased versions thereof being thin wrappers around those that just deal with the phasing aspect.  Not that important, just thinking out loud.

Reply:
I was torn about this. We only added the `Phased` versions with @mathiaswagner 's residency work, and left the un-`Phased` versions for backwards compatibility/`C` not having default arguments. There's no backwards compatibility here, so having a non-phased version would just be confusing... (though maybe no more confusing than having `Phased` flavors of anything in the first place)

Reply:
Sounds fine.  We can leave it as is, and address at a later date if ever we need to.